src/client/customers.rs

```diff
@@ -235,6 +235,203 @@ pub struct Address {
     pub state: Option<String>,
 }
 
+/// The types of ledger entries that can be created.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize)]
+#[serde(tag = "entry_type")]
+pub enum LedgerEntryRequest<'a> {
+    /// Increment a credit balance
+    #[serde(rename = "increment")]
+    Increment(AddIncrementCreditLedgerEntryRequestParams<'a>),
+    /// Void an existing ledger entry
+    #[serde(rename = "void")]
+    Void(AddVoidCreditLedgerEntryRequestParams<'a>),
+    // TODO: additional ledger entry types
+}
+
+/// Optional invoicing settings for a credit purchase.
+#[derive(Debug, Clone, Default, PartialEq, Eq, Hash, Serialize)]
+pub struct CreditLedgerInvoiceSettingsRequestParams<'a> {
+    /// Whether the credits purchase invoice should auto collect with the customer's saved payment
+    /// method.
+    pub auto_collection: bool,
+    /// The difference between the invoice date and the issue date for the invoice. If due on issue,
+    /// set this to `0`.
+    pub net_terms: u64,
+    /// An optional memo to display on the invoice
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub memo: Option<&'a str>,
+}
+
+/// The parameters used to create a customer credit ledger entry.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize)]
+pub struct AddIncrementCreditLedgerEntryRequestParams<'a> {
+    /// The amount to credit the customer for.
+    pub amount: serde_json::Number,
+    /// An optional description for the credit operation.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<&'a str>,
+    /// The date on which the block's balance will expire.
+    #[serde(with = "time::serde::rfc3339::option")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub expiry_date: Option<OffsetDateTime>,
+    /// The date on which the block's balance will become available for use.
+    #[serde(with = "time::serde::rfc3339::option")]
```

Comment:
`chono` uses rfc3339 by default, and in general we should use it instead of `time`, but that can be a followup: https://docs.rs/chrono/latest/chrono/serde/index.html

Reply:
I'd like to go in the other direction, actually! `time` is better maintained than chrono these days. That's why I picked it here. This crate isn't and won't ever be used by the main database so it seemed like a good place to start using `time`.

Reply:
til about chrono vs time switching places (again)